opensearch/src/main/java/org/opensearch/sql/opensearch/planner/physical/ADOperator.java

```diff
@@ -63,22 +66,37 @@ public class ADOperator extends MLCommonsOperatorActions {
   @Override
   public void open() {
     super.open();
-    DataFrame inputDataFrame = generateInputDataset(input);
+    String categoryField = arguments.containsKey(CATEGORY_FIELD)
+        ? (String) arguments.get(CATEGORY_FIELD).getValue() : null;
```

Comment:
avoid using null value?

Reply:
what do you suggest? i thought about optional but code style said to not use `Optional` as function parameters, and other arguments defaults to null https://github.com/opensearch-project/sql/blob/50d8d2da51731f0222ccaff7366ee64c5e4170b8/opensearch/src/main/java/org/opensearch/sql/opensearch/planner/physical/ADOperator.java#L128-L131